sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -427,7 +428,38 @@ public String getDefaultLanguage() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<PiiEntityCollection> recognizePiiEntities(String document, String language) {
-        return recognizePiiEntityAsyncClient.recognizePiiEntities(document, language);
+        return recognizePiiEntityAsyncClient.recognizePiiEntities(document, language, null);
+    }
+
+    /**
+     * Returns a list of Personally Identifiable Information(PII) entities in the provided document
+     * with provided language code.
+     *
+     * For a list of supported entity types, check: <a href="https://aka.ms/tanerpii">this</a>.
+     * For a list of enabled languages, check: <a href="https://aka.ms/talangs">this</a>.
+     *
+     * <p><strong>Code sample</strong></p>
+     * <p>Recognize the PII entities details in a document with provided language code and RecognizePiiEntityOptions.
+     * Subscribes to the call asynchronously and prints out the entity details when a response is received.</p>
+     *
+     * {@codesnippet com.azure.ai.textanalytics.TextAnalyticsAsyncClient.recognizePiiEntities#string-string-RecognizePiiEntityOptions}
+     *
+     * @param document the text to recognize PII entities details for.
+     * For text length limits, maximum batch size, and supported text encoding, see
+     * <a href="https://docs.microsoft.com/azure/cognitive-services/text-analytics/overview#data-limits">data limits</a>.
+     * @param language The 2 letter ISO 639-1 representation of language. If not set, uses "en" for English as default.
+     * @param options The additional configurable {@link RecognizePiiEntityOptions options} that may be passed when
+     * recognizing PII entities.
+     *
+     * @return A {@link Mono} contains a {@link PiiEntityCollection recognized PII entities collection}.
+     *
+     * @throws NullPointerException if {@code document} is null.
+     * @throws TextAnalyticsException if the response returned with an {@link TextAnalyticsError error}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<PiiEntityCollection> recognizePiiEntities(String document, String language,
+        RecognizePiiEntityOptions options) {
```

Comment:
should we check here `Objects.requireNonNull(document, "'document' cannot be null.");` like u did in the non-async client

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/14714/files/f8a5e955184f0573a0ddc2e4a7c475e413a5b85c#diff-ad3e913b93dfdd01bab30e92092a03aaR77

It is checked inside of recognizePiiEntityAsyncClient.recognizePiiEntities(document, language, options) method.

Reply:
seems I did have dup check that ahead of this. I will remove the dup check in sycn client.